app/src/main/java/com/github/se/eventradar/ui/navigation/NavGraph.kt

```diff
@@ -34,8 +36,14 @@ fun NavGraph(navController: NavHostController) {
       context.toast("Profile screen needs to be implemented")
     }
     composable(Route.MY_HOSTING) {
-      HomeScreen(navigationActions = navActions)
-      context.toast("My hosted events screen needs to be implemented")
+      val currentUser = FirebaseAuth.getInstance().currentUser
+      val uid =
+          if (currentUser != null) {
+            currentUser.uid
+          } else {
+            throw IllegalStateException("User is not authenticated")
```

Comment:
Not a good idea to throw an uncaught exception in Android code. We can use null assertion instead. I also think it's better to have this call in the viewModel than in the navigation (but if you think this is better, I'd be curious to know you reasoning).

Reply:
I think that it's better for this call to be in viewModel too rather than NavGraph since only the eventCards rendered will depend on the userId, other parts of the Hosting UI itself should not depend on the userId, therefore it should be passed as a dependency only to the getHostedEvents function. The reason I passed it from navigation was to avoid calling user data from the HostingUI, but it didn't really solve the problem because I still called it in the NavGraph which is still part of UI layer. I think calling the user in the viewModel is a better way to do it such that it prepares all the data for the UI, instead of having the NavGraph do it. Thanks!